index.js

```diff
@@ -89,11 +90,29 @@ const getCompiledFileSync = (moduleDir, path) => {
   return transformSrc(moduleDir, res.toString())
 }
 
+const getBsConfigModuleOptions = buildDir => {
+  const bsconfig = readBsConfigSync(buildDir)
+  if (!bsconfig) {
+    throw new Error(`bsconfig not found in ${buildDir}`)
+  }
+
+  if (!bsconfig['package-specs'] || !bsconfig['package-specs'].length) {
+    throw new Error('package-specs not defined in bsconfig')
+  }
+
+  const moduleSpec = bsconfig['package-specs'][0]
+  const module = typeof moduleSpec === 'string' ? moduleSpec : moduleSpec.module
+  const inSource =
+    typeof moduleSpec === 'string' ? false : moduleSpec['in-source']
+  return { module, inSource }
+}
+
 module.exports = function loader() {
   const options = getOptions(this) || {}
   const buildDir = options.cwd || CWD
-  const moduleDir = options.module || 'js'
-  const inSourceBuild = options.inSource || false
+  const bsconfig = getBsConfigModuleOptions(buildDir)
+  const moduleDir = bsconfig.module || 'js'
```

Comment:
So this isn't a breaking change, can we add the options passed in from the Webpack config as the first choice? Something like:

```js
const moduleDir = options.module || bsconfig.module || 'js'
const inSourceBuild = options.inSource || bsconfig.inSource || false
```

Reply:
Well technically it's not breaking since the options would just be ignored and if your options and bsconfig don't match bs-loader won't work today right?

But I agree having the options there explicitly first is a safer change 👍  I'll add that back

Reply:
Is it a breaking change to make peoples' breaking builds work 🤔 

But yea, I think it's a safer option too

Reply:
One thing about that though is that then the defaults from reason-scripts will stay in place so in-source builds for that wouldn't work. Unless a new version of reason-scripts with this change would just remove the webpack options as well 👍

Reply:
Yea, I think we should push that up on the reason-scripts side